packages/x-charts/src/models/seriesType/line.ts

```diff
@@ -8,6 +8,19 @@ export interface LineSeriesType extends CommonSeriesType, CartesianSeriesType {
   area?: any;
 }
 
+/**
+ * An object that allows to identify a single line.
+ * Used for item interaction
+ */
+export type LineItemIdentifier = {
+  type: 'line';
+  seriesId: LineSeriesType['id'];
+  /**
+   * data Index can be undefined if the mouse is over the area and not a specific item.
+   */
+  dataIndex?: number;
```

Comment:
When the mouse hovers a line area, there is no `dataIndex` because it's the entire series that get the interactions
When the mouse hovers a line mark, there is a `dataIndex`